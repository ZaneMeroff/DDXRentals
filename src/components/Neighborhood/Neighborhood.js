import React, { Component } from 'react';
import './Neighborhood.scss';
import { Link } from 'react-router-dom';


const Neighborhood = props => {
  console.log(props);
  const allListings = props.areas.listings.map(listing => {
    let splitOut = listing.split('/');
    let coolWords = ['Radical', 'Gnarly', 'Tubular', 'Supreme', 'Off-The-Hook', 'Bees-Knees', 'Cats Pajamas', 'Outrageous', 'Awesome', 'Spectacular'];
    let chosenWord = coolWords[Math.floor(Math.random() * 10)];
    return <li className='feature-item' >{chosenWord} Listing {splitOut[4]}</li>
  });
  return (
    <aside className='sidebar-zone'>
      <h1 className='nickname-header'>{props.areas.nickName}</h1>
      <h2>{props.areas.name}</h2>
      <h3>{props.areas.location}</h3>
      <p>{props.areas.about}</p>
      <ul id='area-listings'>
        {allListings}
      </ul>
      <Link to={`/listings`}><button type='button' className='listing-button'>See All Listings</button></Link>
    </aside>
  )
}

export default Neighborhood;
