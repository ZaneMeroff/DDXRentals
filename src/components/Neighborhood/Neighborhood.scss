.sidebar-zone {
  width: 350px;
  position: absolute;
  z-index: 5;
  background-color: #202020;
  top: 80px;
  left: 70%;
  overflow: hidden;
  height: 100%;
  text-align: right;
  padding-right: 50px;
  color: #fff;
  padding-top: 50px;
}

.listing-button {
  height: 40px;
  width: 150px;
  background-color: #DB49CB;
  border: none;
  border-radius: 20px;
  color: #FFF;
  font-size: 1em;
}

.listing-button:hover {
  cursor: pointer;
  background-color: #02A1FF;
  border: none;
}

.nickname-header {
  color: #FFBF03;
  font-size: 2.5em;
}

.feature-item {
  color: #02A1FF;
  list-style: none;
}

#area-listings {
  margin: 6% 4% 6% 4%;
}
