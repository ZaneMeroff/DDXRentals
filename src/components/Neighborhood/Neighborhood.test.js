import React from 'react';
import ReactDOM from 'react-dom';
import Neighborhood from './Neighborhood';
import { shallow } from 'enzyme';
import { mockRandom } from 'jest-mock-random';

describe('Neighborhood', () => {

  let mockAreas;

  beforeEach(() => {
    mockRandom([0.1]);
  });

  it('should be an instance of Neighborhood component', () => {
    mockAreas = {
        id: 0,
        name: 'Select An Area on the Left to View Full Details',
        location: null,
        about: '',
        listings: ['/api/v1/listings/3921']
      }
    const wrapper = shallow(<Neighborhood
      areas={mockAreas}
      />);
      expect(wrapper).toMatchSnapshot();
  });

});
