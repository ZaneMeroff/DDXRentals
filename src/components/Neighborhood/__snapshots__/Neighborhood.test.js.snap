// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Neighborhood should be an instance of Neighborhood component 1`] = `
<aside
  className="sidebar-zone"
>
  <h1
    className="nickname-header"
  />
  <h2>
    Select An Area on the Left to View Full Details
  </h2>
  <h3 />
  <p />
  <ul
    id="area-listings"
  >
    <li
      className="feature-item"
    >
      Gnarly
       Listing 
      3921
    </li>
  </ul>
  <Link
    to="/listings"
  >
    <button
      className="listing-button"
      type="button"
    >
      See All Listings
    </button>
  </Link>
</aside>
`;
